validator/src/main/java/com/google/daq/mqtt/registrar/LocalDevice.java

```diff
@@ -295,25 +304,41 @@ private DeviceCredential getDeviceCredential(String keyFile) throws IOException
   }
 
   private Set<String> keyFiles() {
-    if (!isDirectConnect()) {
+    if (!isDirectConnect() || !hasAuthType()) {
       return ImmutableSet.of();
     }
-    Set<String> publicKeyFile = Set.of(publicKeyFile());
-    Set<String> privateKeyFiles = privateKeyFiles();
-    return Sets.union(publicKeyFile, privateKeyFiles);
+    String authType = getAuthType();
+    Set<String> certFile = getCertFiles();
+    Set<String> publicKeyFile = Set.of(getPublicKeyFile());
+    Set<String> privateKeyFiles = getPrivateKeyFiles();
+    Set<String> rtn = (authType.equals(ES_CERT_TYPE) || authType.equals(RSA_CERT_TYPE)) ? 
```

Comment:
don't abbreviate -- either just return inline, or use semantically meaningful name (not "return" or "rtn")

Reply:
done, I've chosen to return inline